ReSharper.FSharp/src/FSharp.Common/src/Shim/TypeProviders/ExtensionTypingProviderShim.fs

```diff
@@ -38,10 +39,15 @@ type ExtensionTypingProviderShim(solution: ISolution, toolset: ISolutionToolset,
         if isConnectionAlive() then typeProvidersHostLifetime.Terminate()
 
     let connect () =
-        if not (isConnectionAlive ()) then
-            typeProvidersHostLifetime <- Lifetime.Define(lifetime)
-            connection <- typeProvidersLoadersFactory.Create(typeProvidersHostLifetime.Lifetime).Run()
-            typeProvidersManager <- TypeProvidersManager(connection) :?> _
+        if isConnectionAlive () then () else
+
+        lock createProcessLockObj (fun () ->
+
+        if isConnectionAlive () then () else
+
+        typeProvidersHostLifetime <- Lifetime.Define(lifetime)
+        connection <- typeProvidersLoadersFactory.Create(typeProvidersHostLifetime.Lifetime).Run()
+        typeProvidersManager <- TypeProvidersManager(connection) :?> _)

```

Comment:
Please indent lambda body.

Reply:
oh, a situation is possible when the connection is initialized, but the manager is not